update/process/shardBlock.go

```diff
@@ -81,12 +89,14 @@ func (s *shardBlockCreator) CreateNewBlock(
 		return nil, nil, err
 	}
 
+	sftVersionLength := core.MinInt(core.MaxSoftwareVersionLength, len(version))
```

Comment:
You can add in core package a method like "TrimSoftwareVersion(version) trimmedVersion" and use it everywhere you needed

Reply:
done